src/cls/_ZPM/DependencyAnalyzer.cls

```diff
@@ -32,7 +32,7 @@ ClassMethod RunForAllModules() As %Status
 
 /// @API.Method
 /// @Modifier Availability InterSystems
-ClassMethod RunForModule(pModuleName As %String, Output pInvalidReferences, pVerbose As %Boolean = 1, pSave As %Boolean = 1, pProgressListener As %ZPM.PackageManager.Developer.IProgressListener = {$$$NULLOREF}, Output pRunID As %String) As %Status
+ClassMethod RunForModule(pModuleName As %String, Output pInvalidReferences, pVerbose As %Boolean = 1, pSave As %Boolean = 1, Output pRunID As %String) As %Status
```

Comment:
Shouldn't be removed. Should be replaced with %ZHSLIB.PackageManager.Developer.Extension.PackageManager.UI.DependencyAnalyzerSocket.

If that class will be in the separate zpm-studio-ui package, may want to actually add IProgressListener back as %ZPM.General.IProgressListener perhaps.

Reply:
@isc-kiyer Yeah so the UI.DependencyAnalyzerSocket is removed and will not be presented in zpm-studio-ui package as well, because we are not supporting the dependency analyzer studio UI for IPM. In that case, is fine to just delete the IProgressListener here (as the original class is just an abstract cls)? Or actually should wejust remove the ZPM.DependencyAnalyzer class as well since we're not using the dependencyAnalyzer UI?

Reply:
Ah ok in that case, just update references to the method call. If there are currently none in zpm, copy the dependency analyzer command line command we added recently in ZHSLIB so that the dependency analyzer is still available in some form.

Reply:
Also, in that case, can just write out the statements rather than logging them to the progress listener for printing all at once later/in a UI.

Reply:
@isc-kiyer Sounds good, I will put new dependencyAnalyzer calls to %ZPM.Main.cls. Additionally, for calls in this class, in addition to replacing IProgressListener usage with Write statements, I think we still needs to modify those method arguments, since IProgressListener class is now removed, if we still keep its type, the class would report error during build compilation

Reply:
Great! Yes I agree need to remove all references to IProgressListener

Reply:
Add & remove done; new jira here: https://usjira.iscinternal.com/browse/HSIEO-5548